java/compaction-job-execution/src/main/java/sleeper/compaction/jobexecution/RunTasks.java

```diff
@@ -105,67 +178,114 @@ public void run() throws InterruptedException {
         int queueSize = CommonJobUtils.getNumberOfMessagesInQueue(sqsJobQueueUrl, sqsClient)
                 .get(QueueAttributeName.ApproximateNumberOfMessages.toString());
         LOGGER.info("Queue size is {}", queueSize);
+
+        // Obtain details of instances in this cluster
+        Map<String, InstanceDetails> details = InstanceDetails.fetchInstanceDetails(this.clusterName, ecsClient);
+        LOGGER.debug("Cluster container instances {}", details);
+        Set<String> recentContainerInstanceARNs = new HashSet<>();
+
         if (0 == queueSize) {
             LOGGER.info("Finishing as queue size is 0");
-            return;
-        }
+        } else {
```

Comment:
This is a lot of nested blocks now. Could we extract some functions to make it clearer?

Reply:
Done